tests/multiply.apln

```diff
@@ -0,0 +1,154 @@
+        ⍝ come under the region of tolerant equality
+        Hdbl←{⍵,-⍵}1E14+(2×⍳50)                       ⍝ 645: large numbers
+        Sdbl←{⍵,-⍵}(⍳500)÷1000                        ⍝ 645: Small numbers
+
+        ⍝ Hfl is 1287 but larger numbers to test for CT value
+        ⍝ far intervals are chosen for non overlap 
+        ⍝ with region of tolerant equality
+        ⎕FR←fr_decf
+        fl←{⍵,-⍵}i3+0.01                              ⍝ 1287: 128 bits Decimal
+        Hfl←{⍵,-⍵}2E29+(1E16×⍳10)
+        ⎕FR←fr_dbl
+
+        r←⍬
+        testDesc←{'for ',case,{0∊⍴case2:'',⍵⋄' , ', case2,⍵}, '& ⎕FR:', ⎕FR}
+
+        r case case2←⍬ ⍬ ⍬
+
+        :For fr :In 2 1
+            ⎕FR←fr⊃fr_dbl fr_decf
+
+            quadparams←⎕CT ⎕DCT ⎕FR ⎕IO ⎕DIV
+            case←'independant'
+            desc←testDesc⍬
+
+            r,←'TI1' desc Assert 0≡0×0
+            r,←'TNull1' desc Assert ⍬≡''×0 ⍝ testing for null values
+            r,←'TNull2' desc Assert ⍬≡⍬×0
+            r,←'TNull3' desc Assert ⍬≡0×''
+            r,←'TNull4' desc Assert ⍬≡0×⍬
+
+            :For case :In 'bool' 'i1' 'i2' 'i3' 'dbl' 'fl' 'cmplx' 'Hdbl' 'Sdbl' 'Hfl' 'Hcmplx'
+                data←⍎case
+                desc←testDesc⍬
+
+                ⍝ General tests to test the basic rules of multiply
+                r,← 'TGen1' desc Assert (≢data)≡≢data×data
+                
+                d1←(data~0)[?≢data~0]
+
+                r,← 'T1' desc quadparams RunVariations (data model data) data data ⍝ model test
+                d1←data[?≢data]
+                d2←data[?≢(data~d1)]
+                r,← 'T2' desc quadparams RunVariations data data 1         ⍝ Identity
+                r,← 'T3' desc quadparams RunVariations (0⍨¨data) data 0    ⍝ x×0=0
+                r,← 'T4' desc Assert 0≡0×d1
+                r,← 'T5' desc Assert 0≡d1×0
+                r,← 'T6' desc Assert d1≡1×d1
+                r,← 'T7' desc Assert d1≡d1×1
+                r,← 'T8' desc Assert (d1×d2)≡d2×d1              ⍝ commutability
+            :EndFor
+
+            ⍝ Cross type tests
+            :For case2 :In 'i1' 'i2' 'i3' 'dbl' 'fl' 'cmplx' 'Hdbl' 'Hfl' 'Hcmplx'
+                data2←⍎case2
+                data data2←data stripToSameLen data2
+                desc←testDesc⍬
+                :If (case≡case2)
+                    :Continue
+                :EndIf
+
+                r,← 'TCross1' desc quadparams RunVariations (data model data2) data data2 ⍝ check result of different datatypes using model
+                r,← 'TCross2' desc quadparams RunVariations (data2 model data) data2 data ⍝ TCross1 but reversed
+                r,← 'TCross3' desc quadparams RunVariations ((data2,data) model (data,data2)) (data2,data) (data,data2) ⍝ concat 2 different types
+                r,← 'TCross4' desc Assert (((data2 data) model (data data2))≡(data2 data)×(data data2)) ⍝ merge two different arrays creating a pointer array
+
+
+            :EndFor
+
+            ⍝ tests for known errors

```

Comment:
Multiply is a trivial primitive, but I'd still like to see tests for more places where we expect failure. E.g., ×⍨1e155. We don't want this to generate a 1287 accidentally, for example (even though that's improbable, we should still treat the interpreter as a black box and create that test).

Reply:
Want to confirm:
> We don't want this to generate a 1287 accidentally

Should I test if this gives a domain error or not?